core/Pos/Core/Genesis/Parser.hs

```diff
@@ -2,24 +2,29 @@
 -- | Compile-time genesis data parser
 
 module Pos.Core.Genesis.Parser
-       ( compileGenCoreData
+       ( compileGenSpec
        ) where
 
-import           Data.FileEmbed          (embedFile, makeRelativeToProject)
 import           Universum               hiding (lift)
 
-import           Pos.Binary.Class        (decodeFull)
+import           Data.FileEmbed          (embedFile, makeRelativeToProject)
+import           Text.JSON.Canonical     (fromJSON, parseCanonicalJSON)
+import qualified Data.ByteString.Lazy    as BSL
+
 import           Pos.Binary.Core.Genesis ()
 import           Pos.Core.Constants      (genesisBinSuffix)
-import           Pos.Core.Genesis.Types  (GenesisCoreData (..))
+import           Pos.Core.Genesis.Types  (GenesisSpec (..))
 
 -- | Fetch pre-generated genesis data from /genesis-core.bin/ in compile
 -- time. Doesn't use TH with lift because it's difficult to provide 'Lift'
 -- instance to 'GenesisCoreData'
-compileGenCoreData :: GenesisCoreData
-compileGenCoreData =
+compileGenSpec :: GenesisSpec
+compileGenSpec =
+    --let file = $(embedFile =<< makeRelativeToProject ("init-data-" <> genesisBinSuffix <> ".json"))
     let file = $(embedFile =<< makeRelativeToProject ("genesis-core-" <> genesisBinSuffix <> ".bin"))
-    in case decodeFull file of
+    in case parseCanonicalJSON (BSL.fromStrict file) of
```

Comment:
We *could* be more lenient here and parse typical JSON. Canonical JSON is only needed when verifying the genesis in production: the canonical rendering of the parsed thing must hash to some known value, but it doesn't matter if the input JSON has extra whitespace or keys out of alphanumeric order etc.

Reply:
@avieth I know, we decided to compute hash of canonical json instead of input json today, however, I wrote this code earlier